sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -273,12 +271,36 @@ object FileFormatWriter extends Logging {
    * automatically trigger task aborts.
    */
   private trait ExecuteWriteTask {
+
     /**
-     * Writes data out to files, and then returns the list of partition strings written out.
-     * The list of partitions is sent back to the driver and used to update the catalog.
+     * The data structures used to measure metrics during writing.
      */
-    def execute(iterator: Iterator[InternalRow]): Set[String]
+    protected val writingTimePerFile: mutable.ArrayBuffer[Long] = mutable.ArrayBuffer.empty
```

Comment:
Now I exclude the zero values of writing time (actually < 1ms). So I actually don't count totoalWritingTime / numFiles.

totoalWritingTime / numFiles can easily result a zero writing time, for example the total writing time for (1ms, 1ms, 0ms, 0ms) is 2ms, and totoalWritingTime / numFiles = 2ms / 4 = 0ms (the smallest unit is 1ms).